pkg/odo/cli/dev/dev.go

```diff
@@ -0,0 +1,136 @@
+package dev
+
+import (
+	"fmt"
+	dfutil "github.com/devfile/library/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/redhat-developer/odo/pkg/devfile/adapters/kubernetes"
+	"github.com/redhat-developer/odo/pkg/envinfo"
+	"github.com/redhat-developer/odo/pkg/odo/cli/component"
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions/clientset"
+	odoutil "github.com/redhat-developer/odo/pkg/odo/util"
+	"github.com/redhat-developer/odo/pkg/util"
+	"github.com/spf13/cobra"
+	"k8s.io/kubectl/pkg/util/templates"
+	"os"
+	"path/filepath"
+)
+
+// RecommendedCommandName is the recommended command name
+const RecommendedCommandName = "dev"
+
+type DevOptions struct {
+	// Context
+	*genericclioptions.Context
+
+	// Flags
+	contextFlag string
+
+	// Clients
+	clientset *clientset.Clientset
+
+	// Variables
+	ignorePaths []string
```

Comment:
@feloy wdyt about making `ignorePaths` a part of `DevOptions` struct vs. evaluating it right before calling:
```go
err = o.clientset.DevClient.Start(o.Context.EnvSpecificInfo.GetDevfileObj(), platformContext, o.ignorePaths, path, os.Stdout)
```
While writing this, I feel that it should not be a part of struct unless we are going to provide a `--ignore` flag with `odo dev`. Since we are evaluating it in odo logic, it would be better to evaluate right before the call.